### PR TITLE
refactor(utils): rewrite partition

### PR DESCRIPTION
**Pull request by fisker**
use Array#reduce, make partition generic

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If not an internal change) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
why no .editorconfig file in this project

File: src/utils/partition.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-      a.push(item);
-    } else {
-      b.push(item);
-    }
-  });
-  return [a, b];
+  return reduce.call(
+    array,
+    (result, item) => {
+      result[fn(item) ? 0 : 1].push(item);
```
Comment (azz) [inline src/utils/partition.js]: I personally find this harder to read.

**Comment by lydell** [comment]
Thank you for your contribution! I think we’re happy with the way the function is already implemented, though. Since this does not fix any bug, let’s not spend time on it. If you’re looking for ways to contribute, there are plenty of issues labeled [bug](https://github.com/prettier/prettier/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Atype%3Abug) – find one that looks interesting and see if you can fix it!

File: src/utils/partition.js (hunk 1 above)
Comment (fisker) [reply src/utils/partition.js]: reduce is much reasonable,  and `lodash` has the exactly same implementation

https://github.com/lodash/lodash/blob/d5ef31929a1262abbc75b8dadc0b6ae6e9558b5f/partition.js#L26

**Pull request closed without merging**